### PR TITLE
Extra checks

### PR DESCRIPTION
**Pull request by frankmullenger**
Checks for session cookies and cache headers in responses, introduces Guzzle dependency.

### CODE REVIEW

**Comment by frankmullenger** [comment]
Happy to remove the return type declarations if Travis builds for earlier PHP versions are going to remain in the module for now?

**Comment by ScopeyNZ** [comment]
> Happy to remove the return type declarations if Travis builds for earlier PHP versions are going to remain in the module for now?

Yeah that'd be great. Have to wait a few more months before we can start using PHP7 specific features.

**Comment by NightJar** [comment]
That was going to be my feedback, yeah :)

**Comment by NightJar** [comment]
`phpcs` @frankmullenger

**Comment by frankmullenger** [comment]
Not sure what the policy is on merging with failed code coverage tests but if these need unit tests that is probably a job for another hack day :-)

**Comment by NightJar** [comment]
Ideally they should exist, but on the other hand they could be a follow up.
I'll leave it open for colleagues to evaluate. I think tests would be bests.

File: src/Traits/Fetcher.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    /**
+     * Fetch a response for a URL using Guzzle client.
+     *
+     * @param string $url
+     * @param array|null $extraConfig Extra configuration
+     * @return ResponseInterface
+     */
+    public function fetchResponse(string $url, array $extraConfig = [])
```
Comment (robbieaverill) [inline src/Traits/Fetcher.php]: Scalar typehints aren't PHP 5.6 friendly =)

**Comment by frankmullenger** [comment]
Fair enough, will look to add some unit tests next hackday 👍

**Pull request merged**